### PR TITLE
Sleep improvements

### PR DESCRIPTION
**Pull request by SoneeJohn**
**Fix:**
* 	Fix issue that caused assertion to be created multiple times
screen shot 2016-11-16 at 8 48 59 pm

**Notable Changes:**
* Will only show sleep menu items on client. The iMac will not show these options. "While in Target Display Mode, the host iMac automatically ignores any scheduled sleep commands and keeps the system running as long as the source’s video signal is flowing. If your source device sleeps, however, it will break the connection and the host iMac will revert to the internal display." - https://www.tekrevue.com/tip/ins-outs-imacs-target-display-mode/

* `Allow Display To Sleep` will prevent the client from going into idle sleep.
* `Allow System To Sleep` will prevent the client's display from going to sleep.
*  `Auto Toggle Display` will not hidden on the client side.

Client View:
screen shot 2016-11-16 at 9 17 13 pm

Host View:
screen shot 2016-11-16 at 9 16 17 pm

I'm still testing this particularly in regards to having `Allow Display To Sleep` on. I want to test if I can have the iMac's display *off* while still being connected to my rMBP. If I wake my rMBP I want to see if the iMac's display will *turn* back on. Note that the machine will be clamshell mode.

**Update**: Just tested it and it works. On the client side, having none of the sleep options check, when the MacBook when display turned off the iMac would exit TDM. However, when I checked `Allow Display To Sleep` and the Macbook's display turned off TDM, was still active but the iMac's display was *off*. Once, I woke my computer (via the keyboard) the iMac's display lit up and the MacBook's display showed instead of the iMac's display.

**Update 2:** I think I might want to change the sleep menu items to just one: `Allow Display To Sleep`. If `Allow Display To Sleep` is not checked then the app will hold `kIOPMAssertPreventUserIdleDisplaySleep`.

### CODE REVIEW

**Comment by duanefields** [comment]
I don't have a way to test the client any more so just let me know when you think it's ready to merge.

**Comment by reitermarkus** [comment]
> I think I might want to change the sleep menu items to just one: `Allow Display To Sleep`.

This is a good idea, since display sleep also implies system sleep, because of course the system can't sleep without the display sleeping.

I would however switch the behaviour to be `Prevent Display Sleep`.

File: VirtualKVM/KVMController.m [hunk 1]
```diff
@@ -207,9 +207,9 @@ - (void)enableTargetDisplayMode {
     IOReturn success = IOPMAssertionCreateWithName(assertionType, kIOPMAssertionLevelOn, reasonForActivity, &_sleepAssertion);
 
     if (success == kIOReturnSuccess) {
-      NSLog(NSLocalizedString(@"Sleep disabled.", comment:nil));
+      NSLog(NSLocalizedString(@"Created power assertion. Assertion type: %@", nil),assertionType);
```
Comment (reitermarkus) [inline VirtualKVM/KVMController.m]: Seems like `comment:` got missing here.

File: VirtualKVM/KVMController.m [hunk 2]
```diff
@@ -185,27 +189,39 @@ - (void)enableTargetDisplayMode {
...
+  
 }
 
 - (void)disableTargetDisplayMode {
   if (self.sleepAssertion != kIOPMNullAssertionID) {
     IOReturn success = IOPMAssertionRelease(self.sleepAssertion);
 
     if (success == kIOReturnSuccess) {
-      NSLog(NSLocalizedString(@"Sleep enabled.", comment:nil));
+      NSLog(NSLocalizedString(@"Release power assertion.", comment:nil));
```
Comment (reitermarkus) [inline VirtualKVM/KVMController.m]: Release**d**

**Comment by SoneeJohn** [comment]
@reitermarkus done!

**Comment by SoneeJohn** [comment]
screen shot 2016-11-22 at 5 39 15 pm
Updated client screenshot.

**Comment by duanefields** [comment]
Ready to merge?

**Comment by SoneeJohn** [comment]
@donaldchai Yeah I'm ready to merge.

**Comment by jlipshutz** [comment]
I was looking for this functionality but it doesn't seem to be in the latest build. How can I access this?

**Pull request merged**